### PR TITLE
Improve Site Settings Permission checks

### PR DESCRIPTION
**Pull request by infofromca**

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.GitHub/Drivers/GithubAuthenticationSettingsDisplayDriver.cs [hunk 1]
```diff
@@ -63,7 +63,8 @@ public override async Task<IDisplayResult> EditAsync(GitHubAuthenticationSetting
         }
 
         public override async Task<IDisplayResult> UpdateAsync(GitHubAuthenticationSettings settings, BuildEditorContext context)
-        {
+        {           
+
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.GitHub/Drivers/GithubAuthenticationSettingsDisplayDriver.cs:67]: No need for extra line

File: src/OrchardCore.Modules/OrchardCore.Email/Drivers/SmtpSettingsDisplayDriver.cs [hunk 2]
```diff
@@ -39,7 +39,7 @@ public override async Task<IDisplayResult> EditAsync(SmtpSettings settings, Buil
         {
             var user = _httpContextAccessor.HttpContext?.User;
 
-            if (!await _authorizationService.AuthorizeAsync(user, Permissions.ManageEmailSettings))
+            if (user == null || !await _authorizationService.AuthorizeAsync(user, Permissions.ManageEmailSettings))
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Email/Drivers/SmtpSettingsDisplayDriver.cs:42]: Why the null check? Doesn't work previously?!!

File: src/OrchardCore.Modules/OrchardCore.Twitter/Drivers/TwitterSettingsDisplayDriver.cs [hunk 3]
```diff
@@ -68,7 +68,7 @@ public override async Task<IDisplayResult> EditAsync(TwitterSettings settings, B
         }
 
         public override async Task<IDisplayResult> UpdateAsync(TwitterSettings settings, BuildEditorContext context)
-        {
+        {
```
Comment (hishamco) [inline src/OrchardCore.Modules/OrchardCore.Twitter/Drivers/TwitterSettingsDisplayDriver.cs:71]: Extra whitespace

**Review by hishamco — CHANGES_REQUESTED** [review]
Could you please describe what the issue you are trying to fix

File: src/OrchardCore.Modules/OrchardCore.Email/Drivers/SmtpSettingsDisplayDriver.cs (hunk 2 above)
Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Email/Drivers/SmtpSettingsDisplayDriver.cs:42]: There should be no need I agree with @hishamco

Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Email/Drivers/SmtpSettingsDisplayDriver.cs:42]: https://github.com/OrchardCMS/OrchardCore/blob/454105788cd2e9184c15a620eb3c212ad30cb488/src/OrchardCore/OrchardCore.Infrastructure/Security/AuthorizationServiceExtensions.cs#L17-L20

**Review by hishamco — CHANGES_REQUESTED** [review]
LGTM, but it would be great to see the final changes ONLY that are made in this PR to avoid unpredictable results ;)

**Review by deanmarcussen — APPROVED** [review]

**Pull request merged**